streams/src/test/java/org/apache/kafka/streams/integration/StandbyTaskCreationIntegrationTest.java

```diff
@@ -105,32 +106,81 @@ public void close() {}
             }, stateStoreName);
 
         final Topology topology = builder.build();
-        client1 = new KafkaStreams(topology, streamsConfiguration());
-        client2 = new KafkaStreams(topology, streamsConfiguration());
+        createClients(topology, streamsConfiguration(), topology, streamsConfiguration());
+
+        setStateListenersForVerification(thread -> thread.standbyTasks().isEmpty() && !thread.activeTasks().isEmpty());
+
+        startClients();
+
+        waitUntilBothClientAreOK(
+            "At least one client did not reach state RUNNING with active tasks but no stand-by tasks"
+        );
+    }
+
+    @Test
+    public void shouldCreateStandByTasksForMaterializedAndOptimizedSourceTables() throws Exception {
+        final Properties streamsConfiguration1 = streamsConfiguration();
+        streamsConfiguration1.put(StreamsConfig.TOPOLOGY_OPTIMIZATION, StreamsConfig.OPTIMIZE);
+        final Properties streamsConfiguration2 = streamsConfiguration();
+        streamsConfiguration2.put(StreamsConfig.TOPOLOGY_OPTIMIZATION, StreamsConfig.OPTIMIZE);
```

Comment:
Probably not that important, but are these two Properties objects exactly the same?

Reply:
No, they are not because of `streamsConfiguration.put(StreamsConfig.STATE_DIR_CONFIG, TestUtils.tempDirectory(applicationId).getPath());`. Each instance of the `Properties` has a different state directory, which is good because the test does not work otherwise. I tried it out.